hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/OmUtils.java

```diff
@@ -527,4 +529,45 @@ public static void validateKeyName(String keyName)
               OMException.ResultCodes.INVALID_KEY_NAME);
     }
   }
+
+  /**
+   * Return configured OzoneManager service id based on the following logic.
+   * Look at 'ozone.om.internal.service.id' first. If configured, return that.
+   * If the above is not configured, look at 'ozone.om.service.ids'.
+   * If count(ozone.om.service.ids) == 1, return that id.
+   * If count(ozone.om.service.ids) > 1 throw exception
+   * If 'ozone.om.service.ids' is not configured, return null. (Non HA)
+   * @param conf configuration
+   * @return OM service ID.
+   * @throws IOException on error.
+   */
+  public static String getOzoneManagerServiceId(OzoneConfiguration conf)
+      throws IOException {
+    Collection<String> omServiceIds;
+    String localOMServiceId = conf.get(OZONE_OM_INTERNAL_SERVICE_ID);
+    if (localOMServiceId == null) {
+      LOG.info("{} is not defined, falling back to {} to find serviceID for "
+              + "OzoneManager if it is HA enabled cluster",
+          OZONE_OM_INTERNAL_SERVICE_ID, OZONE_OM_SERVICE_IDS_KEY);
+      omServiceIds = conf.getTrimmedStringCollection(
+          OZONE_OM_SERVICE_IDS_KEY);
+      if (omServiceIds.size() > 1) {
+        throw new IOException(String.format(
+            "More than 1 OzoneManager ServiceID (ozone.om.service.ids) " +
+                "configured : %s, but ozone.om.internal.service.id is not " +
+                "configured.", omServiceIds.toString()));
+      }
+    } else {
+      omServiceIds = Collections.singletonList(localOMServiceId);
+    }
+
```

Comment:
I think we also need to check if localServiceID value is one of the ozone.om.service.ids value.
To find if there is a mistake in the config.

Reply:
Thanks @bharatviswa504. Will fix this.